Fastpack/Error.ml

```diff
@@ -36,20 +83,34 @@ let to_string package_dir error =
       filename
 
   | CannotResolveModule (path, dep) ->
-    let dep_str = Module.Dependency.to_string ~dir:(Some package_dir) dep in
+    let dep_str = Module.Dependency.to_string ~dir:(Some package_dir) dep in 
     Printf.sprintf
       "\n%s\nWhile processing dependency request:\n\t%s\n"
       path
       dep_str
 
-  | CannotParseFile (location_str, errors) ->
-    let format_error (loc, error) =
-      loc_to_string loc ^ " " ^ FlowParser.Parse_error.PP.error error
+  | CannotParseFile (location_str, errors, source) ->
+    let lines = String.split_on_char '\n' source
+                |> List.mapi (fun i line -> (i + 1, line)) in
+    let format_error (loc, error) = 
+      let error_desc = FlowParser.Parse_error.PP.error error in 
+      "--------------------\n"
+      ^ error_desc ^ " at " ^ (loc_to_string loc) ^ "\n"
+      ^ "\n"
+      ^ (get_codeframe loc lines)
+      ^ "\n"
```

Comment:
You can use the `String.concat` in cases like this one: 
```OCaml
String.concat "\n" [
  "--------------------";
  error_desc ^ " at " ^ loc_to_string loc;
  get_codeframe loc lines;
  ""
]
```

Reply:
I applied this trick in a couple places.  The empty strings look a bit funny, but definitely still a readability boost :)